optuna/integration/allennlp.py

```diff
@@ -15,12 +18,166 @@
     import allennlp
     import allennlp.commands
     import allennlp.common.util
-    from allennlp.training import EpochCallback
 
+# EpochCallback is conditionally imported because allennlp may be unavailable in
+# the environment that builds the documentation.
 if _imports.is_successful():
     import _jsonnet
+    from allennlp.training import EpochCallback
 else:
-    EpochCallback = object  # NOQA
+    # I disable mypy here since `allennlp.training.EpochCallback` is a subclass of `Registrable`
+    # (https://docs.allennlp.org/master/api/training/trainer/#epochcallback) but `EpochCallback`
+    # defined here is not `Registrable`, which causes a mypy checking feailure.
+    class EpochCallback:  # type: ignore
+        """Stub for EpochCallback."""
+
+        @classmethod
+        def register(cls: Any, *args: Any, **kwargs: Any) -> Callable:
+            """Stub method for `EpochCallback.register`.
+
+            This method has the same signature as
+            `Registrable.register <https://docs.allennlp.org/master/
+            api/common/registrable/#registrable>`_ in AllenNLP.
+
+            """
+
+            def wrapper(subclass: Any, *args: Any, **kwargs: Any) -> None:
+                return subclass
+
+            return wrapper
+
+
+_PPID = os.getppid()
+
+"""
+User might want to launch multiple studies that uses `AllenNLPExecutor`.
+Because `AllenNLPExecutor` uses environment variables for communicating
+between a parent process and a child process. A parent process creates a study,
+defines a search space, and a child process trains a AllenNLP model by
+`allennlp.commands.train.train_model`. If multiple processes use `AllenNLPExecutor`,
+the one's configuration could be loaded in the another's configuration.
+To avoid this hazard, we add ID of a parent process to each key of
+environment variables.
+"""
+PREFIX = "{}_OPTUNA_ALLENNLP".format(_PPID)
+MONITOR = "{}_MONITOR".format(PREFIX)
+PRUNER_CLASS = "{}_PRUNER_CLASS".format(PREFIX)
+PRUNER_KEYS = "{}_PRUNER_KEYS".format(PREFIX)
+STORAGE_NAME = "{}_STORAGE_NAME".format(PREFIX)
+STUDY_NAME = "{}_STUDY_NAME".format(PREFIX)
+TRIAL_ID = "{}_TRIAL_ID".format(PREFIX)
```

Comment:
Can we make these variables private similar to `_PPID`?
```suggestion
_PREFIX = "{}_OPTUNA_ALLENNLP".format(_PPID)
_MONITOR = "{}_MONITOR".format(PREFIX)
_PRUNER_CLASS = "{}_PRUNER_CLASS".format(PREFIX)
_PRUNER_KEYS = "{}_PRUNER_KEYS".format(PREFIX)
_STORAGE_NAME = "{}_STORAGE_NAME".format(PREFIX)
_STUDY_NAME = "{}_STUDY_NAME".format(PREFIX)
_TRIAL_ID = "{}_TRIAL_ID".format(PREFIX)
```

Reply:
Sorry for not considering the suggestion https://github.com/optuna/optuna/pull/1772#discussion_r488378436 well. 🙇 
I made these constants private in 41e654f.